frontend/src/pages/projects/projectSharing/ProjectSharingTable.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import Table from '~/components/Table';
+import { RoleBindingKind } from '~/k8sTypes';
+import { deleteRoleBinding, generateRoleBindingProjectSharing, createRoleBinding } from '~/api';
+import ProjectSharingTableRow from './ProjectSharingTableRow';
+import { columnsProjectSharingUser, columnsProjectSharingGroup } from './data';
+import { ProjectSharingRBType } from './types';
+
+type ProjectSharingTableProps = {
+  type: ProjectSharingRBType;
+  permissions: RoleBindingKind[];
+  typeAhead: string[];
+  onCancel: () => void;
+  refresh: () => void;
+};
+
+const ProjectSharingTable: React.FC<ProjectSharingTableProps> = ({
+  type,
+  permissions,
+  typeAhead,
+  onCancel,
+  refresh,
+}) => {
+  const [editCell, setEditCell] = React.useState<string[]>([]);
+
+  return (
+    <Table
+      variant="compact"
+      data={permissions}
+      columns={
+        type === ProjectSharingRBType.USER ? columnsProjectSharingUser : columnsProjectSharingGroup
+      }
+      disableRowRenderSupport
+      rowRenderer={(rb) => (
+        <ProjectSharingTableRow
+          key={rb.metadata?.name || ''}
+          obj={rb}
+          isEditing={rb.subjects[0]?.name === '' || editCell.includes(rb.metadata.name)}
+          typeAhead={typeAhead}
+          onCreateOrEditRoleBinding={(name, roleType, obj) => {
+            const newRBObject = generateRoleBindingProjectSharing(
+              obj.metadata.namespace,
+              type,
+              name,
+              roleType,
+            );
+            if (obj.subjects[0]?.name === '') {
+              createRoleBinding(newRBObject)
+                .then(() => refresh())
+                .catch(() =>
+                  setEditCell((prev) => prev.filter((cell) => cell !== rb.metadata.name)),
```

Comment:
@kywalker-rh  this is the error-handling, we might wanna define how can we manage the errors here.

Reply:
Hmmm... maybe an error bubble on the right side of the line with an expanded popup for the error details? Seems the space doesn't really offer it up... could toast on save => error 🤔

Reply:
Happy to discuss what options we have @lucferbux. I'm assuming this shows up if for some reason we can't add a user (they don't exist, issue talking to OpenShift, etc).

Reply:
Well, this is rolebindings, so you can have pretty much any name you want regardless if the user exists or not. That might be useful if some admin want to configure a project ahead of the user creation/admission for example (that's one of the items I left underlayed in the description of the PR)

But yes, errors might happen, the cluster might be down or there might be permission issues for some operations, that sort of things, it would be nice to say the user "hey, your changes haven't committed, that why they are not. present in the table" right now it just magically refresh the table.

Reply:
Agreed, we should tell the user something. I think a dismissible inline alert where the row would otherwise appear is the right implementation. Let me know if you need me to mock something up @lucferbux

Reply:
Right, I can add that, if I don't see the layout clear I'll ping you but it seems pretty straightforward thanks! @kywalker-rh

Reply:
![fail-state](https://user-images.githubusercontent.com/16117276/233416660-a1bc4e5b-4bdd-4ec2-b3e6-5acc680d9413.gif)

This is the end result